### PR TITLE
feat(ingest/snowflake): support for more operation types

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by treff7es — APPROVED** [review]
lgtm

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py [hunk 1]
```diff
@@ -43,6 +43,20 @@
     "CREATE": OperationTypeClass.CREATE,
     "CREATE_TABLE": OperationTypeClass.CREATE,
     "CREATE_TABLE_AS_SELECT": OperationTypeClass.CREATE,
+    "MERGE": OperationTypeClass.CUSTOM,
+    "COPY": OperationTypeClass.CUSTOM,
+    "TRUNCATE_TABLE": OperationTypeClass.CUSTOM,
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:48]: I see that putting `CUSTOM` allows us to show the `customOperationType`, but for other sources we try to label these:

(unity)
```
    QueryStatementType.COPY: OperationTypeClass.INSERT,
    QueryStatementType.UPDATE: OperationTypeClass.UPDATE,
    QueryStatementType.MERGE: OperationTypeClass.UPDATE,
    QueryStatementType.DELETE: OperationTypeClass.DELETE,
    QueryStatementType.TRUNCATE: OperationTypeClass.DELETE,
```

(bigquery)
```
    "MERGE": OperationTypeClass.UPDATE,
```

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py [hunk 2]
```diff
@@ -363,6 +379,9 @@ def _get_operation_aspect_work_unit(
                     lastUpdatedTimestamp=last_updated_timestamp,
                     actor=user_urn,
                     operationType=operation_type,
+                    customOperationType=query_type
+                    if operation_type is OperationTypeClass.CUSTOM
+                    else None,
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:384]: Is there harm in passing this if the operation type class is not `CUSTOM`, for more info? I know we don't do this elsewhere in the code but could start here

**Review by asikowitz — APPROVED** [review]
Saw this just got merged, but a couple question

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:48]: MERGE is in fact UPDATE and/or DELETE and/or INSERT, so I was hesitant to mark it as UPDATE. CUSTOM operationType is not great to use. cc: @jjoyce0510 . We could add more operation types in model maybe ?  I am proposing addition of more operation types to  - MERGE, COPY, TRUNCATE respectively. Does that make sense ?

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:384]: hmm, I thought that myself as well. Then customOperationType just behaves more like `"nativeOperationType"` . cc - @jjoyce0510  - can you confirm this one as well please ?

**Pull request merged**